### ISSUE TITLE
[State] Register and expose ABIs via inbuilt chain state/storage

### ISSUE THREAD

**Issue by compleatang** [hyperledger/burrow#858]
a new field should be added to the `deploy` job which should be a boolean titled `register` (or some synonym of that) -- it should default to `false`.

the result of `register: true` should register the ABI of a contract in the name registry of burrow with the following convention in pseudocode:

```
name: abi.$(sha256(bytecode)[:40]
value: $(json.Stringify(ABI))
```

**Comment by compleatang** [comment]
issue migrated from https://github.com/monax/bosmarmot/issues/106

**Comment by phymbert** [comment]
+1 on this feature request. Additionnaly it would be great to add a helper method (for example in Transactor API) to call contract and map function inputs/outputs automatically to go types:
```go
var returnValues []interface{} = transactor.Call(callTx, "contractFunctionName", arg1, arg2, argX)
```

**Comment by silasdavis** [comment]
We should look at https://github.com/ethereum/EIPs/issues/820

**Comment by silasdavis** [comment]
@seanyoung has this nearly completed!

**Comment by silasdavis** [comment]
fixed by #1125
